service/header/sync.go

```diff
@@ -0,0 +1,120 @@
+package header
+
+import (
+	"context"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+// syncer implements simplest possible synchronization for headers.
+type syncer struct {
+	exchange Exchange
+	store    Store
+
+	// done is triggered when syncing is finished
+	// it assumes that syncer is only used once
+	done chan struct{}
+}
+
+func newSyncer(exchange Exchange, store Store) *syncer {
+	return &syncer{
+		exchange: exchange,
+		store:    store,
+		done:     make(chan struct{}),
+	}
+}
+
+// Sync syncs all headers up to the latest known header in the network.
+func (s *syncer) Sync(ctx context.Context) {
+	log.Info("syncing headers")
+	// TODO(@Wondertan): Retry logic
+	for {
+		localHead, err := s.store.Head(ctx)
+		if err != nil {
+			log.Errorw("getting local head", "err", err)
+			return
+		}
+
+		netHead, err := s.exchange.RequestHead(ctx)
+		if err != nil {
+			log.Errorw("requesting network head", "err", err)
+			return
+		}
+
+		if localHead.Height >= netHead.Height {
+			// we are now synced
+			close(s.done)
+			log.Info("synced headers")
+			return
+		}
+
+		err = s.syncDiff(ctx, localHead, netHead)
+		if err != nil {
+			log.Errorw("syncing headers", "err", err)
+			return
+		}
+	}
+}
+
+// TODO(@Wondertan): This is totally random number that I feel like should work for splitting big requests
+//  into smaller
+var requestSize uint64 = 100
```

Comment:
Is this 100 bytes? Or headers? I assume headers, then:
```suggestion
// TODO(@Wondertan): Number of headers that can be requested at once. Either make this configurable or, 
// find a proper rationale for constant.
var requestSize uint64 = 128
```